HeterogeneousCore/CUDAUtilities/test/copyAsync_t.cpp

```diff
@@ -40,8 +41,9 @@ TEST_CASE("copyAsync", "[cudaMemTools]") {
 
       SECTION("Copy all") {
         cudautils::copyAsync(device, host_orig, N, stream);
-        cuda::memory::async::copy(host.get(), device.get(), N * sizeof(int), stream);
         cudaCheck(cudaStreamSynchronize(stream));
+	cudaMemcpyAsync(host.get(), device.get(), N *  sizeof(int), cudaMemcpyDeviceToHost, stream);
```

Comment:
These two lines seem incorrect:
  - `cudaMemcpyAsync` should come before `cudaStreamSynchronize`
  - `stream.synchronize();` seems spurious, possibly due to incorrect conflict resolution